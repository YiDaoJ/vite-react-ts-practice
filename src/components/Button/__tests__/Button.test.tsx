import React from "react";

import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Button } from "..";

describe("Button test", () => {
  test("render Button component", () => {
    const handleClick = jest.fn();
    const { asFragment } = render(
      <Button size="large" onClick={handleClick}>
        Test
      </Button>
    );
    // screen.debug();

    expect(asFragment()).toMatchSnapshot();
  });

  test("button click event", async () => {
    const handleClick = jest.fn();
    render(
      <Button size="large" onClick={handleClick}>
        Test
      </Button>
    );
    const element = screen.getByRole("button");

    await userEvent.click(element);
    expect(handleClick).toHaveBeenCalled();
  });
});
