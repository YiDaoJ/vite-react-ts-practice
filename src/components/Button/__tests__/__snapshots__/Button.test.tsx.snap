// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test render Button component 1`] = `
<DocumentFragment>
  <button
    class="css-83quec-Button-getSize e1afpsjd0"
  >
    Test
  </button>
</DocumentFragment>
`;
